### PR TITLE
Use registerAuthenticationChangedHandler

### LINKED ISSUE #136: Use registerAuthenticationChangedHandler
`registerAuthenticationChangedHandler` method introduced
which should replace KO specific subscription to observables notifications like `authManager.authenticated.subscrbe(...)`
This is change required to start using interfaces instead of global `authManager`

### PR DESCRIPTION
**Pull request by mtutynina**
resolve #136

### CODE REVIEW

**Comment by kant2002** [comment]
There somewhere should be also `authManager.login.subscribe`. In essence this could be replaced with registerAuthenticationChangedHandler too.

File: tests/table/playerCards.test.ts [hunk 1]
```diff
@@ -153,8 +154,10 @@ describe("Player cards", function () {
                 Money: 10000,
             }];
             const actions: GameActionStartInformation[] = [];
+            authenticated(false);
```
Comment (kant2002) [inline tests/table/playerCards.test.ts]: Why this change?

**Review by kant2002 — CHANGES_REQUESTED** [review]

File: tests/table/playerCards.test.ts (hunk 1 above)
Comment (mtutynina) [reply tests/table/playerCards.test.ts]: Because subscribe notifies only if value is not the same as previous, in some testst authenticated already was true, so we should "logout" first and then "login"

Comment (kant2002) [reply tests/table/playerCards.test.ts]: How this value could be initially true if we never set it to true previously

Comment (mtutynina) [reply tests/table/playerCards.test.ts]: i don't know, but without this some tests doesn't work

Comment (kant2002) [reply tests/table/playerCards.test.ts]: Your tests are interfere with each other, since we use global `authManager`. So value of `authenticated` kept true from previous tests. In order to make controlled tests use `beforeAll`, set `authenticated(false)` in that method, so each test would start with not authenticated user.

**Review by kant2002 — APPROVED** [review]
